src/Fable.Transforms/Fable2Babel.fs

```diff
@@ -1261,16 +1261,20 @@ module Util =
         | Fable.NewAnonymousRecord(values, fieldNames, _genArgs, _isStruct) ->
             let values = List.mapToArray (unwrapOptionalArg com >> snd >> transformAsExpr com ctx) values
             Array.zip fieldNames values |> makeJsObject
-        | Fable.NewUnion(values, tag, ent, genArgs) ->
-            let ent = com.GetEntity(ent)
+        | Fable.NewUnion(values, tag, entRef, genArgs) ->
+            let ent = com.GetEntity(entRef)
             if com.IsTypeScript then
-                let case = ent.UnionCases[tag]
-                match tryJsConstructorWithSuffix com ctx ent ("_" + case.Name) with
-                | Some helperRef when not(UnionHelpers.caseNameClashes case.Name)->
-                    let values = values |> List.mapToArray (transformAsExpr com ctx)
-                    let typeParams = makeTypeParamInstantiation com ctx genArgs
-                    Expression.callExpression(helperRef, values, typeArguments=typeParams)
-                | _ -> transformNewUnionForTypeScript com ctx r ent tag values case
+                match List.tryItem tag ent.UnionCases with
+                | Some case ->
+                    match tryJsConstructorWithSuffix com ctx ent ("_" + case.Name) with
+                    | Some helperRef when not(UnionHelpers.caseNameClashes case.Name)->
```

Comment:
Given that you are adding the check we can move the `UnionHelpers.caseNameClashes case.Name` up too.

```
match List.tryItem tag ent.UnionCases with
| Some case when not(UnionHelpers.caseNameClashes case.Name) ->
```